test/e2e/specs/onboarding/setup__site-assembler.ts

```diff
@@ -26,9 +26,17 @@ describe( DataHelper.createSuiteTitle( 'Site Assembler' ), () => {
 	} );
 
 	describe( 'Create a site with the Site Assembler', function () {
-		let assembledPreviewLocator: Locator;
+		let assembledPreviewFrameLocator: FrameLocator;
 		let startSiteFlow: SiteAssemblerFlow;
 
+		const getLargePreviewFrameLocator = () => {
+			if ( ! assembledPreviewFrameLocator ) {
+				assembledPreviewFrameLocator = page.frameLocator( '.pattern-large-preview iframe' );
+			}
+
+			return assembledPreviewFrameLocator;
```

Comment:
**comment** - this is a design pattern that I haven't seen in our E2Es yet. Typically, the values are assigned objects, not a function. 

Since a frameloctor is retrieved synchronously, it's safe to simply assign it at compile time like this:

```suggestion
    const assembledPreviewFrameLocator = page.frameLocator('.pattern-large-preview iframe'')
    
    // when using
    
    expect(assembledPreviewFrameLocator
					.locator( '.wp-site-blocks > *[data-type="header"]' )
					.count()
			).toBe( 1 );
```

Reply:
> we cannot get it at the beginning since the iframe is shown when we have any selected pattern

Just to clarify, the iframe does not show at first but begins to show once a component is selected? 

Even if that’s the case, the FrameLocator will resolve once a component is selected. Since the check is after a component is selected, my guess is that it should be fine to simply define the locator and expect it to resolve upon being called. 

I’ll try to check on my side as well.

Reply:
Interesting! I tried it and it works 👍 I've also updated the code